extensions/wikia/WallNotifications/tests/WallNotificationEntityTest.php

```diff
@@ -72,55 +75,81 @@ private function getWallMessageMock( array $params ) {
 			->method( 'getWallOwner' )
-			->willReturn( $wallOwner );
-		$wm->expects( $this->once() )
+			->willReturn( $wallOwnerMock );
+
+		$wallMessageMock->expects( $this->at( $wallMessageCounter++ ) )
 			->method( 'getArticleTitle' )
-			->willReturn( $articleTitle );
-		$wm->expects( $this->once() )
-			->method( 'getTitle' )
-			->willReturn( $title );
-		$wm->expects( $this->once() )
-			->method( 'getTopParentObj' )
-			->willReturn( $parentWm );
-		$wm->expects( $this->once() )
+			->willReturn( $articleTitleMock );
+
+		$wallMessageMock->expects( $this->at( $wallMessageCounter++ ) )
 			->method( 'getText' )
 			->willReturn( $params['wallMessage']['text'] );
-		$wm->expects( $this->once() )
+
+		$wallMessageMock->expects( $this->at( $wallMessageCounter++ ) )
+			->method( 'getTitle' )
+			->willReturn( $titleMock );
+
+		$wallMessageMock->expects( $this->at( $wallMessageCounter++ ) )
 			->method( 'getMessagePageUrl' )
 			->willReturn( $params['wallMessage']['messagePageUrl'] );
-		$wm->expects( $this->once() )
+
+		$wallMessageMock->expects( $this->at( $wallMessageCounter++ ) )
 			->method( 'getNotifyeveryone' )
 			->willReturn( $params['wallMessage']['notifyeveryone'] );
-		$wm->expects( $this->once() )
+
+		$isEdited = $params['wallMessage']['isEdited'];
+		$wallMessageMock->expects( $this->at( $wallMessageCounter++ ) )
 			->method( 'isEdited' )
-			->willReturn( $params['wallMessage']['isEdited'] );
-		$wm->expects( $this->any() )
-			->method( 'getLastEditSummary' )
-			->willReturn( $params['wallMessage']['lastEditSummary'] );
-		$wm->expects( $this->any() )
-			->method( 'getMetaTitle' )
-			->willReturn( $params['wallMessage']['metaTitle'] );
+			->willReturn( $isEdited );
+
+		if ( $isEdited ) {
+			$wallMessageMock->expects( $this->at( $wallMessageCounter++ ) )
+				->method( 'getLastEditSummary' )
+				->willReturn( $params['wallMessage']['lastEditSummary'] );
+		}
+
+		if ( !empty( $params['parentWallMessage']['id'] ) ) {
+			$parentWallMessageMock = $this->getParentWallMessageMock( $params );
+
+			$wallMessageMock->expects( $this->at( $wallMessageCounter++ ) )
+				->method( 'getTopParentObj' )
+				->willReturn( $parentWallMessageMock );
+		} else {
+			$wallMessageMock->expects( $this->at( $wallMessageCounter++ ) )
+				->method( 'getTopParentObj' )
+				->willReturn( null );
+
+			$wallMessageMock->expects( $this->at( $wallMessageCounter++ ) )
+				->method( 'getMetaTitle' )
+				->willReturn( $params['wallMessage']['metaTitle'] );
+		}
 
-		return $wm;
+		$this->mockClass( WallMessage::class, $wallMessageMock, 'newFromTitle' );
 	}
 
 	/**
 	 * @param array $params
 	 *
 	 * @return PHPUnit_Framework_MockObject_MockObject|Revision
 	 */
-	private function getRevisionMock( array $params ) {
+	private function getRevisionMock( array $params ): PHPUnit_Framework_MockObject_MockObject {
 		$rev = $this->getMockBuilder( 'Revision' )
 			->setMethods( [
 				'getId',
 				'getUser',
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "getUser" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)